kmk/scanners/bidirectional.py

```diff
@@ -0,0 +1,229 @@
+import digitalio
+
+from kmk.matrix import KeyEvent, intify_coordinate as ic
+from kmk.scanners import Scanner
+
+
+def mapping_up_down(cols, rows):
+    '''
+    Maps keys according to the following diagram:
+
+       1 2 3
+     +------
+    1| A A A
+    2| A A A
+     |
+    1| B B B
+    2| B B B
+    '''
+
+    coord_mapping = []
+    for y in range(2 * rows):
+        coord_mapping.extend(ic(y, x, cols) for x in range(cols))
```

Comment:
A heads-up:
`intify_coordinate` should be regarded as legacy for backwards compatibility . It's easier to understand what's going on if you do:
```python
coord_mapping = list(range(2*cols*rows))
```